scripts/xe-backup-metadata

```diff
@@ -120,8 +103,8 @@ fi
 test_sr "${sr_uuid}"
 
 sr_name=$(${XE} sr-param-get uuid="${sr_uuid}" param-name=name-label)
-# see if a backup VDI already exists on the selected SR
-vdi_uuid=$(${XE} vdi-list other-config:ctxs-pool-backup=true sr-uuid="${sr_uuid}" params=uuid --minimal)
+# assume use of the new format predictable UUID
+vdi_uuid=$(${XE} vdi-list uuid="$(uuid5 "e93e0639-2bdb-4a59-8b46-352b3f408c19" "$sr_uuid")" --minimal)
```

Comment:
The quoting doesn't appear to be correct here, there appears to be a nested " inside an "

Reply:
This is valid - inside the outer quotes there is a `$( ... )`, which the shell does the right thing with.

Reply:
We have too many layers of quoting and languages here (both python and shell).
`sr_uuid` is trusted in this case, but if it wasn't then there'd be a python injection vulnerability here, e.g. this prints "I was here" (and could run arbitrary python code):

```sh
#!/bin/bash
function uuid5 {
    python -c "import uuid; print (uuid.uuid5(uuid.UUID('$1'), '$2'))"
}

sr_uuid="')); print('I was here');(('"

echo uuid="$(uuid5 "e93e0639-2bdb-4a59-8b46-352b3f408c19" "$sr_uuid")"
```

Just mentioning this here that this pattern shouldn't be copied elsewhere, the way arguments are passed to the python code is not safe, it is only safe in this instance because it is coming directly from the output of a trusted `xe` command.